dev/bots/test.dart

```diff
@@ -245,224 +251,159 @@ Future<void> _runBuildTests() async {
+  print('${green}Testing Dart2JS build$reset for $cyan$relativePathToApplication$reset...');
+  await runCommand(flutter,
+    <String>['build', 'web', '-v', '--target=$target'],
+    workingDirectory: path.join(flutterRoot, relativePathToApplication),
+    expectNonZeroExit: expectNonZeroExit,
+    environment: <String, String>{
+      'FLUTTER_WEB': 'true',
+    },
   );
-  print('Done.');
 }
 
-Future<void> _runTests() async {
+Future<void> _runAddToAppTests() async {
+  if (Platform.isMacOS) {
+    print('${green}Running add-to-app iOS integration tests$reset...');
+    final String addToAppDir = path.join(flutterRoot, 'dev', 'integration_tests', 'ios_add2app');
+    await runCommand('./build_and_test.sh',
+      <String>[],
+      workingDirectory: addToAppDir,
+    );
+  }
+}
+
+Future<void> _runFrameworkTests() async {
   final bq.BigqueryApi bigqueryApi = await _getBigqueryApi();
-  await _runSmokeTests();
-  final String subShard = Platform.environment['SUBSHARD'];
 
   Future<void> runWidgets() async {
+    print('${green}Running packages/flutter tests for$reset: ${cyan}test/widgets/$reset');
     await _runFlutterTest(
       path.join(flutterRoot, 'packages', 'flutter'),
+      options: <String>['--track-widget-creation'],
       tableData: bigqueryApi?.tabledata,
-      tests: <String>[
-        path.join('test', 'widgets') + path.separator,
-      ],
+      tests: <String>[ path.join('test', 'widgets') + path.separator ],
     );
-    // Only packages/flutter/test/widgets/widget_inspector_test.dart really
-    // needs to be run with --track-widget-creation but it is nice to run
-    // all of the tests in package:flutter with the flag to ensure that
-    // the Dart kernel transformer triggered by the flag does not break anything.
     await _runFlutterTest(
       path.join(flutterRoot, 'packages', 'flutter'),
-      options: <String>['--track-widget-creation'],
+      options: <String>['--no-track-widget-creation'],
       tableData: bigqueryApi?.tabledata,
-      tests: <String>[
-        path.join('test', 'widgets') + path.separator,
-      ],
+      tests: <String>[ path.join('test', 'widgets') + path.separator ],
     );
+    // Try compiling code outside of the packages/flutter directory with and without --track-widget-creation
+    await _runFlutterTest(path.join(flutterRoot, 'examples', 'flutter_gallery'), options: <String>['--track-widget-creation'], tableData: bigqueryApi?.tabledata);
+    await _runFlutterTest(path.join(flutterRoot, 'examples', 'flutter_gallery'), options: <String>['--no-track-widget-creation'], tableData: bigqueryApi?.tabledata);
   }
 
-  Future<void> runFrameworkOthers() async {
+  Future<void> runLibraries() async {
     final List<String> tests = Directory(path.join(flutterRoot, 'packages', 'flutter', 'test'))
       .listSync(followLinks: false, recursive: false)
       .whereType<Directory>()
       .where((Directory dir) => dir.path.endsWith('widgets') == false)
```

Comment:
If we ever add a new directory that matches `packages/flutter/test/**/widgets`, its tests will silently never be run...

Reply:
unless i misunderstand, i don't believe that's an issue; `recursive` is false

Reply:
Ah, I missed the recursive:false - nevermind.